pulp_2to3_migration/app/pre_migration.py

```diff
@@ -479,17 +479,20 @@ def pre_migrate_distributor(repo_id, distributors, distributor_migrators, repo=N
                 distributor.pulp2_config = dist_data.config
                 distributor.pulp2_last_updated = last_updated
                 distributor.is_migrated = False
+                distributor.save()
                 dist_migrator = distributor_migrators.get(distributor.pulp2_type_id)
                 needs_new_publication = dist_migrator.needs_new_publication(distributor)
                 needs_new_distribution = dist_migrator.needs_new_distribution(distributor)
                 remove_publication = needs_new_publication and distributor.pulp3_publication
                 remove_distribution = needs_new_distribution and distributor.pulp3_distribution
+
                 if remove_publication:
+                    # TODO check when publication shared by multiple distro
+                    # FIXME when checksum_type is added to the publication
                     distributor.pulp3_publication.delete()
                 if remove_publication or remove_distribution:
                     distributor.pulp3_distribution.delete()
 
-            distributor.save()
```

Comment:
Why does it need to be moved?
What happens when distributor hasn't been updated but is just reiadded to the plan (aka doesn't go into `if` statement, but  `distributor.not_in_plan = False` needs to be set)

Reply:
good point about not_in_plan flag i missed that one

```
In [14]: d.pulp3_distribution                                                                                                                                                                                      
Out[14]: <BaseDistribution (pulp_type=file.file): 5edf9114c998ac34be0d66db-file-many>

In [15]: d.pulp3_distribution.delete()                                                                                                                                                                             
Out[15]: (2, {'file.FileDistribution': 1, 'core.BaseDistribution': 1})

In [16]: d.save()                                                                                                                                                                                                  
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-16-ad0d43fe2a3d> in <module>
----> 1 d.save()

/usr/local/lib/pulp/lib64/python3.6/site-packages/django/db/models/base.py in save(self, force_insert, force_update, using, update_fields)
    691                     raise ValueError(
    692                         "save() prohibited to prevent data loss due to "
--> 693                         "unsaved related object '%s'." % field.name
    694                     )
    695                 # If the relationship's pk/to_field was changed, clear the

ValueError: save() prohibited to prevent data loss due to unsaved related object 'pulp3_distribution'.
```